### ISSUE TITLE
Compatibility with NSBatchDeleteRequest & NSBatchUpdateRequest

### ISSUE THREAD

**Issue by kishanmadhu** [project-imas/encrypted-core-data#235]
Hi,

I am using EncryptedIncrementalStore to use encryption with core data. I want to delete some records from one entity using NSBatchDeleteRequest, When I fetching the result, I am getting nil error in response.

Following method supports only NSFetchRequestType & NSSaveRequestType
```
- (id)executeRequest:(NSPersistentStoreRequest *)request
         withContext:(NSManagedObjectContext *)context
               error:(NSError **)error
```

How can I use NSBatchDeleteRequest & NSBatchUpdateRequest?

Thanks.

**Comment by ppaulojr** [comment]
Yes, that's very important. I just spend several hours troubleshooting `NSBatchDeleteRequest` before realizing the problem is `EncryptedCoreData` not supporting it..

**Comment by ghost** [comment]
This feature would make life easier. Hope it will be in one of the next releases.

**Comment by Dennis-Cl** [comment]
I also found the same problem in our project, now I have to fetch all the records and iterator to delete each object  in context, the performance will be a big problem!!!
